### PR TITLE
Recipe for bustools (formerly #16042)

### PR DESCRIPTION
**Pull request by apeltzer**
This is @astrovsky01 s work to get the recipe up on bioconda. I just merged his PR to this branch to get things going (and others, including me) can help in adding tests and missing bits and pieces 👍 

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by apeltzer** [comment]
@astrovsky01 @bgruening this should be ready to go now 👍

File: recipes/bustools/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    name: bustools
+    version: {{ version }}
+
+source:
+    url: https://github.com/BUStools/bustools/archive/v{{ version }}.tar.gz
+    sha256: f42184fdd20309b1f77d09e97e8f9005b81ab030eb924ce69f6c62cedd50ce57
+
+build:
+    number: 0
+    string: BUStools
```
Comment (bgruening) [inline recipes/bustools/meta.yaml]: why is this "string" needed?

File: recipes/bustools/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+requirements:
+    build:
+        - {{ compiler('cxx') }}
+        - cmake
+    run:
+        - hdf5
+        - zlib
+    host:
```
Comment (bgruening) [inline recipes/bustools/meta.yaml]: both runtime deps do have run_exports defined and can be removed.

File: recipes/bustools/meta.yaml (hunk 1 above)
Comment (apeltzer) [reply recipes/bustools/meta.yaml]: Its not - I just removed it :-)

File: recipes/bustools/meta.yaml (hunk 2 above)
Comment (apeltzer) [reply recipes/bustools/meta.yaml]: Just removed that as well

**Comment by astrovsky01** [comment]
Wow, thanks @apeltzer, I had been intending on finishing this up this morning. I appreciate it!

**Comment by bgruening** [comment]
Cool, thanks a bunch!

**Comment by apeltzer** [comment]
You did the work in the beginning - thanks a lot for starting this @astrovsky01 ! That's the nice part I love about bioconda's work. 

Thanks for reviewing too @bgruening !

**Pull request merged**